File: design-system/tokens/Tokens.swift
// Ruum Ruum — Design Tokens v1.0 (Swift)
// Fuente: design-system/tokens/tokens.json
// Uso: Text("Hola").font(.ruumBody).foregroundColor(.ruumNavy)
import SwiftUI

public extension Color {
    static let ruumNavy = Color(hex: "0A2342")
    static let ruumTeal = Color(hex: "00D1D1")
    static let ruumTealDeep = Color(hex: "008B8B")
    static let ruumAction = Color(hex: "0066FF")
    static let ruumSurface = Color(hex: "E8EEF4")
    static let ruumMuted = Color(hex: "566889")
    static let ruumBorderInput = Color(hex: "7A8DAE")
    static let ruumBorder = Color(hex: "E6F0FF")
    static let ruumWarning = Color(hex: "F5B400")
    static let ruumWarningBg = Color(hex: "FFF4D6")
    static let ruumWarningText = Color(hex: "7A4F00")
    static let ruumSuccess = Color(hex: "16805A")
    static let ruumSuccessBg = Color(hex: "E6F6EE")
    static let ruumSuccessText = Color(hex: "0F6B48")
    static let ruumError = Color(hex: "C23648")
    static let ruumErrorBg = Color(hex: "FDECEF")
    static let ruumEmergency = Color(hex: "B3261E")
    static let ruumNeutralBg = Color(hex: "EEF2F8")
    static let ruumActionBg = Color(hex: "E8F0FE")

    init(hex: String) {
        var h = hex.trimmingCharacters(in: .whitespacesAndNewlines).replacingOccurrences(of: "#", with: "")
        if h.count == 6 { h += "FF" }
        var v: UInt64 = 0
        Scanner(string: h).scanHexInt64(&v)
        self.init(
            .sRGB,
            red: Double((v & 0xFF000000) >> 24) / 255,
            green: Double((v & 0x00FF0000) >> 16) / 255,
            blue: Double((v & 0x0000FF00) >> 8) / 255,
            opacity: Double(v & 0x000000FF) / 255
        )
    }
}

public enum RuumTokens {
    public static let touch: CGFloat = 44
    public static let touchStreet: CGFloat = 56
    public static let buttonHeight: CGFloat = 52
    public static let buttonHeightStreet: CGFloat = 56
    public static let inputHeight: CGFloat = 48
    public static let radiusSM: CGFloat = 8
    public static let radiusMD: CGFloat = 12
    public static let radiusLG: CGFloat = 16
    public static let radiusXL: CGFloat = 24
}

File: design-system/tokens/tokens.ts
/**
 * Ruum Ruum — Design Tokens v1.0 para React Native / TypeScript
 * Fuente: design-system/tokens/tokens.json
 * Uso: import { tokens } from "./tokens"; StyleSheet.create({ btn: { height: tokens.size.buttonHeight } })
 */

export const tokens = {
  color: {
    navy: "#0A2342",
    teal: "#00D1D1",
    tealDeep: "#008B8B",
    blue: "#0066FF",
    action: "#0066FF",
    surface: "#E8EEF4",
    white: "#FFFFFF",
    muted: "#566889",
    borderInput: "#7A8DAE",
    border: "#E6F0FF",
    warning: "#F5B400",
    success: "#16805A",
    error: "#C23648",
    emergency: "#B3261E",
    neutralBg: "#EEF2F8",
    actionBg: "#E8F0FE",
    actionText: "#1461E0",
    warningBg: "#FFF4D6",
    warningText: "#7A4F00",
    successBg: "#E6F6EE",
    successText: "#0F6B48",
    errorBg: "#FDECEF",
    errorText: "#C23648",
  },
  gradient: {
    cta: ["#008B8B", "#0066FF"] as const,
    route: ["#00D1D1", "#0066FF"] as const,
  },
  font: {
    family: "Inter",
    fallback: ["Arial", "Calibri", "sans-serif"] as const,
    size: {
      display: 40,
      h1: 32,
      h2: 24,
      h3: 18,
      body: 16,
      small: 14,
      caption: 12,
      streetBody: 17,
    },
    weight: {
      regular: "400",
      medium: "500",
      semibold: "600",
      bold: "700",
      extrabold: "800",
    } as const,
  },
  space: { s4: 4, s8: 8, s12: 12, s16: 16, s20: 20, s24: 24, s32: 32, s40: 40, s48: 48 },
  radius: { sm: 8, md: 12, lg: 16, xl: 24, full: 999 },
  elevation: {
    e1: { shadowColor: "#0D2B5E", shadowOpacity: 0.05, shadowRadius: 10, shadowOffset: { width: 0, height: 2 } },
    e2: { shadowColor: "#0D2B5E", shadowOpacity: 0.1, shadowRadius: 24, shadowOffset: { width: 0, height: 8 } },
  },
  size: { touch: 44, touchStreet: 56, buttonHeight: 52, buttonHeightStreet: 56, inputHeight: 48 },
  motion: { micro: 150, transition: 250, easing: "ease-out" as const },
} as const;

export type Tokens = typeof tokens;
